external/fv3fit/tests/test_derived_model.py

```diff
@@ -54,7 +59,29 @@ def test_derived_alert_to_missing_additional_input():
 def test_invalid_derived_output_variables():
     with pytest.raises(ValueError):
         DerivedModel(
-            base_model,
-            additional_input_variables=["surface_diffused_shortwave_albedo"],
-            derived_output_variables=["variable_not_in_DerivedMapping"],
+            base_model, derived_output_variables=["variable_not_in_DerivedMapping"],
         )
+
+
+def test_dump_and_load(tmpdir):
+    derived_model = DerivedModel(
+        base_model, derived_output_variables=["net_shortwave_sfc_flux_derived"],
+    )
+    ds_in = xr.Dataset(
+        data_vars={
+            "input": xr.DataArray(np.zeros([3, 3, 5]), dims=["x", "y", "z"],),
+            "surface_diffused_shortwave_albedo": xr.DataArray(
+                np.zeros([3, 3]), dims=["x", "y"],
+            ),
+        }
+    )
+    input_data = ds_in.stack(sample=["x", "y"])
+    prediction = derived_model.predict(input_data)
+
+    fv3fit.dump(derived_model, str(tmpdir))
+    loaded_model = fv3fit.load(str(tmpdir))
+
+    prediction_after_load = loaded_model.predict(input_data)
+    for var in loaded_model.output_variables:
+        assert var in derived_model.output_variables
+        assert np.allclose(prediction[var].values, prediction_after_load[var].values)
```

Comment:
I think we have a utility function in vcm (or just in xarray) that will compare datasets including metadata. You may want to switch to that since a common issue with dumping and loading models is the loaded model can't restore metadata properly.

Reply:
Thanks, switched to use `xr.Dataset.identical`